### PR TITLE
caveats for ngVue and a quirk mode

### PR DESCRIPTION
**Pull request by dorayx**
- [x] a new section for caveats
- [x] a quirk mode to fix the limitation of the reactivity system
- [x] integration tests for the quirk mode
- [x] introduction to the quirk mode

### CODE REVIEW

File: docs/caveats.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+- `push()`
+- `pop()`
+- `shift()`
+- `unshift()`
+- `sort()`
+- `reverse()`
+
+So you can manipulate the array by calling those wrapped methods to trigger the view updates. In comparison, the non-mutating methods, such as `filter()`, `concat()` and `slice()`, provide a reactive immutability -- don't worry about the performance issue, VueJS doesn't throw away the existing DOM but *implements some smart heuristics to maximize DOM element reuse*[[1](https://vuejs.org/v2/guide/list.html#Replacing-an-Array)].
+
+```
```
Comment (nicolaspayot) [inline docs/caveats.md]: `javascript` missing after the three ```

File: docs/plugins.md [hunk 2]
```diff
@@ -108,4 +112,27 @@ angular.module('custom.plugin', ['ngVue.plugins'])
...
+- set an array element with the index: `array[0] = newElement`
+- modify the length of the array: `array.length = 0`
+
+You have to mutate the scope object in a reactive way to trigger the view updates in VueJS. You are likely to refactor all the controller code. So we introduce the quirk mode.
+
+The quirk mode enables AngularJS to propagate all the watched changes to VueJS and so you will not have to refactor any controller code to trigger the reactivity system to re-render the components.
+
+You can activate it with `$ngVueProvider` during the config phase:
+
+```
```
Comment (nicolaspayot) [inline docs/plugins.md]: `javascript` missing after the three ```

File: example/plugins-filters/index.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import angular from 'angular'
 import Vue from 'vue'
 import '../../src/index.js'
```
Comment (nicolaspayot) [inline example/plugins-filters/index.js]: What about `import '../../src/' ?

File: example/plugins-filters/index.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
-import '../../src/plugins.js'
+import '../../src/plugins/index.js'
```
Comment (nicolaspayot) [inline example/plugins-filters/index.js]: What about `import '../../src/plugins/` ?

**Comment by dorayx** [comment]
@nicolaspayot fixed 😄

**Pull request merged**